ProcessLib/Process.h

```diff
@@ -200,6 +203,10 @@ class Process
 
     VectorMatrixAssembler _global_assembler;
 
+    /// Coupled processes.
+    mutable std::weak_ptr<StaggeredCouplingTerm> _coupling_term;
```

Comment:
Maybe it's a stupid question, but does a process couple always with only one other process? If there would be more than one other process to couple with, shouldn't the `_coupling_term` be a vector of weak_ptr's?